### PR TITLE
A few improvements of TerminateSteadyState

### PR DESCRIPTION
**Pull request by devmotion**
The PR contains a few suggestions with possible improvements for TerminateSteadyState.

### CODE REVIEW

File: src/terminatesteadystate.jl [hunk 1]
```diff
@@ -2,6 +2,11 @@
 # Terminate when all derivatives fall below a threshold or
 #   when derivatives are smaller than a fraction of state
 function allDerivPass(integrator, abstol, reltol, min_t)
+    # Early exit
+    if min_t !== nothing && integrator.t < min_t
```
Comment (devmotion) [inline src/terminatesteadystate.jl:6]: No need to perform all evaluations below in this case.

File: src/terminatesteadystate.jl [hunk 2]
```diff
@@ -16,21 +21,25 @@ function allDerivPass(integrator, abstol, reltol, min_t)
     end
 
     if integrator.u isa Array
-        any(abs(d) > abstol && abs(d) > reltol * abs(u)
+        any(abs(d) > abstol || abs(d) > reltol * abs(u)
```
Comment (devmotion) [inline src/terminatesteadystate.jl:24]: According to the docstring the default check ensures that
> all derivatives should become smaller than `abstol` and the states times
`reltol`

File: src/terminatesteadystate.jl [hunk 3]
```diff
@@ -16,21 +21,25 @@ function allDerivPass(integrator, abstol, reltol, min_t)
...
 
-    if min_t === nothing
-        return true
-    else
-        return integrator.t >= min_t
-    end
+    return true
+end
+
+struct WrappedTest{T,A,R,M}
```
Comment (devmotion) [inline src/terminatesteadystate.jl:36]: I think the `let`-trick leads to a bit ugly code 😄

File: src/terminatesteadystate.jl [hunk 4]
```diff
@@ -62,15 +71,14 @@ the [Steady State Solvers](https://docs.sciml.ai/DiffEqDocs/stable/solvers/stead
...
+        min_t = nothing, wrap_test::Val{WT} = Val(true)) where {T,WT}
     condition = if WT
-        (u, t, integrator) -> test(integrator, abstol, reltol, min_t)
+        WrappedTest(test, abstol, reltol, min_t)
     else
         test
     end
-    affect! = (integrator) -> terminate!(integrator)
-    DiscreteCallback(condition, affect!; save_positions = (true, false))
+    DiscreteCallback(condition, terminate!; save_positions = (true, false))
```
Comment (devmotion) [inline src/terminatesteadystate.jl:81]: No need for a separate `affect!` function.

File: src/terminatesteadystate.jl (hunk 2 above)
Comment (devmotion) [reply src/terminatesteadystate.jl:24]: Maybe better to change the docstring though, to be consistent with `isapprox`.

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Thanks, all simple but good changes. Any tangible performance difference from the iteration change?

**Pull request merged**